src/Illuminate/Cache/CacheServiceProvider.php

```diff
@@ -15,27 +15,18 @@ class CacheServiceProvider extends ServiceProvider implements DeferrableProvider
      */
     public function register()
     {
-        $this->app->singleton('cache', function ($app) {
-            return new CacheManager($app);
-        });
+        $this->app->singleton('cache', fn ($app) => new CacheManager($app));
 
-        $this->app->singleton('cache.store', function ($app) {
-            return $app['cache']->driver();
-        });
+        $this->app->singleton('cache.store', fn ($app) => $app['cache']->driver());
 
-        $this->app->singleton('cache.psr6', function ($app) {
-            return new Psr16Adapter($app['cache.store']);
-        });
+        $this->app->singleton('cache.psr6', fn ($app) => new Psr16Adapter($app['cache.store']));
 
-        $this->app->singleton('memcached.connector', function () {
-            return new MemcachedConnector;
-        });
+        $this->app->singleton('memcached.connector', new MemcachedConnector);
```

Comment:
You're changing the behavior here from deferring the creation of the connector to immediate instantiation. Please re-add the closure (or short closure)